interfaces/devdoc/file_requirements.md

```diff
@@ -0,0 +1,123 @@
+# file
+================
+
+## Overview
+
+The `file` module provides a platform-independent API for asynchronous file operations.
+
+-`file_create`: returns a file handle for the given file name.
+-`file_destroy`: closes the given file handle.
+-`file_write_async`: enqueues an asynchronous write request for a file at a given position.
+-`file_read_async`: enqueues an asynchronous read request for a file at a given position and size.
+-`file_extend_filesize`: expands the given file to be of desired size.
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, FILE_HANDLE, file_create,FILE_REPORT_FAULT, user_report_fault_callback, void*, user_report_fault_context, EXECUTION_ENGINE_HANDLE, execution_engine, const char*, full_file_name);
+MOCKABLE_FUNCTION(, void, file_destroy, FILE_HANDLE, handle);
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_WRITE_ASYNC_RESULT, file_write_async, FILE_HANDLE, handle, CONSTBUFFER_HANDLE, source, uint64_t, position, FILE_WRITE_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+MOCKABLE_FUNCTION_WITH_RETURNS(, FILE_READ_ASYNC_RESULT, file_read_async, FILE_HANDLE, handle, uint32_t, size, uint64_t, position, FILE_READ_CB, user_callback, void*, user_context)(0, MU_FAILURE);
+MOCKABLE_FUNCTION_WITH_RETURNS(, int, file_extend_filesize, FILE_HANDLE, handle, uint64_t, desired_size, bool, has_manage_volume)(0, MU_FAILURE);
+```
+
+## file_create
+
+```c
+MOCKABLE_FUNCTION(, FILE_HANDLE, file_create,FILE_REPORT_FAULT, user_report_fault_callback, void*, user_report_fault_context, EXECUTION_ENGINE_HANDLE, execution_engine, const char*, full_file_name);
+```
+
+`file_create` opens an existing file by the name of `full_file_name` or creates a file of `desired_file_size` if it doesn't exist and returns the file handle.

```

Comment:
poetry: Maybe separate sentences because this could be read two ways: (opens existing or (creates a file and returns)) vs. ((opens existing or creates a file) and returns). Something about English grammar not having operator precedence...